### ISSUE TITLE
Probe failed with error -1

### ISSUE THREAD

**Issue by cbalducci** [porjo/mt7601#7]
Hi porjo,
 i have been able to succesfully compile the module on a banana pi running fedora 21

- model name: ARMv7 Processor rev 4 (v7l)
- 3.18.3-201.fc21.armv7hl armv7l
- gcc version 4.9.2 
- gcc target: armv7hl-redhat-linux-gnueabi
- glibc-2.20-7.fc21.armv7hl

but the module loading returns "rt2870: probe of 3-1:1.0 failed with error -1". After this the wifi dongle does not work and nothing more gets logged anywhere.

Compilation procedure, loading error, iwconfig and ip output > http://pastebin.com/Jm6Zi3bu
Compilation stdout > http://pastebin.com/eyiwfJ8g
Compilation stderr (warnings) > http://pastebin.com/8KKwUfGV

do you have any idea how i can go on debugging this issue?
Thank you

**Comment by porjo** [comment]
What does the output from `lsusb` show? For example, my dongle shows up as follows:
```
Bus 001 Device 007: ID 148f:7601 Ralink Technology, Corp. MT7601U Wireless Adapter
```

**Comment by porjo** [comment]
@cbalducci I will close this now. If you are still experiencing problems, just update this issue.

**Comment by cbalducci** [comment]
@porjo sorry, i was travelling

Bus 001 Device 002: ID 148f:7601 Ralink Technology, Corp. MT7601U Wireless Adapter

**Comment by porjo** [comment]
Hmm, OK so it is definitely one of the supported devices.... I'm using a dongle with the same hardware signature on fc21 using kernel 3.18.3-201.fc21.x86_64 and it works fine!?

Do you have another PC running Linux that you could try it with?

**Comment by cbalducci** [comment]
I should be able to check it tomorrow on a fc21 x86_64 laptop. I will keep you posted

**Comment by cbalducci** [comment]
The same device works on 3.17.8-300.fc21.x86_64 and 3.18.3-201.fc21.x86_64

stderr: http://pastebin.com/fizmXb4E
stdout: http://pastebin.com/Y0xQti7J
GCC: 4.9.2-1
GLibC: 2.20-7

lsusb outputs:

Bus 001 Device 007: ID 148f:7601 Ralink Technology, Corp. MT7601U Wireless Adapter

I will check on my raspberry pi tomorrow to be sure that the problem is not in the banana's usb ports (i have tested both). It would seem strange but you never know..

**Comment by blacksad77** [comment]
The same with Pogoplug Mobile: armv5tel, 3.19.0-1-ARCH 
Bus 001 Device 007: ID 148f:7601 Ralink Technology, Corp. MT7601U Wireless Adapter

Dmesg:
[ 4857.949744] usb 1-1.3: new high-speed USB device number 7 using orion-ehci
[ 4858.098929] rt2870: probe of 1-1.3:1.0 failed with error -1
I works on Fedora 21 on x86...

**Comment by kuba-moo** [comment]
You can try setting **RTDebugLevel** to **RT_DEBUG_LOUD** in *src/os/linux/rt_linux.c* on line 54 to get more information in dmesg.

**Comment by blacksad77** [comment]
Sorry for the delay. I did as you told me - result is still the same:
[ 5513.106021] usb 1-1: USB disconnect, device number 4
[ 5517.902556] usb 1-1: new high-speed USB device number 5 using orion-ehci
[ 5518.071350] rt2870: probe of 1-1:1.0 failed with error -1

**Comment by kuba-moo** [comment]
Did you recompile the driver after making the suggested change in the code?

**Comment by blacksad77** [comment]
Of course. It was set to trace earlier, so it should log  more things - i
think debug is commented out in the code somewhere else...

2015-03-09 12:37 GMT+01:00 Jakub Kicinski <notifications@github.com>:

**Comment by erwinkendo** [comment]
I have the same problem, any suggestion?
rt2870: probe of 1-1.2:1.0 failed with error -1

**Comment by kuba-moo** [comment]
Try tracing it perhaps? Something like:
`trace-cmd record -p function insmod ./path_to_the_module.ko`
You need to install the trace-cmd command obviously. If you get this to work we will have a clue about which function fails with that generic -1 error code.

**Comment by erwinkendo** [comment]
No response in this matter?

**Comment by porjo** [comment]
@erwinkendo did you perform the trace as @kuba-moo suggested? Put the output on pastebin and include the link here.

Alternatively, you might like to try the new driver that kuba-moo has created - https://github.com/kuba-moo/mt7601u (you need kernel >= 3.19 for that driver).

**Comment by erwinkendo** [comment]
Hi @porjo , I haven't been able to use trace-cmd in my linux distribution, is there any alternative to this?
I am aware of the new driver but I need AP support and the new driver lacks this functionality.

**Comment by milidam** [comment]
Hi,
I also tried to cross-compile this driver for arm, and encountered the same error when trying to load it on my target: rt2870: probe of 4-1:1.0 failed with error -1.

After a bit of debugging, I found out that it is due to a memory allocation problem of the HTTxContext (trace: "ERROR in Alloc Bulk buffer for HTTxContext!", in common/cmm_mac_usb.c).
This context is a HTTX_BUFFER struct, with a WirelessPacket element of size MAX_TXBULK_SIZE = LOCAL_TXBUF_SIZE*BULKAGGRE_SIZE.

Changing the LOCAL_TXBUF_SIZE #define value in rtmp_def.h from 2048 to 256 does the trick: the driver loads properly.
It seems however a bit unstable, as I loose the connection frequently... but that is maybe another thing (I use it in ad-hoc mode, with wpa_supplicant, and transfer a large amount of bytes...).

I am not a specialist of such kind of driver, so would there be another/better solution?

**Comment by erwinkendo** [comment]
It is true, the reason it fails (at least in my case) is memory allocation error.
I managed to solve it changing the coherent_pool parameter for my embedded system from 25KB to 2M. I think the same method can be used for other systems.

@milidam how do you manage to put the driver in ad-hoc mode and transmit data? have you managed to put the driver in AP mode?

**Comment by milidam** [comment]
@erwinkendo To change the mode, you can simply use `iwconfig ra0 mode <mode>`, where `<mode>` can be `Ad-Hoc`, `Managed` (client), or ''Master'' (AP).
Note that the driver has to support the mode... luckily for me, the mt7601 driver supports the ad-hoc mode :)

**Comment by porjo** [comment]
FYI, kernel 4.2 is now stable. It includes an mt7601u driver. I recommend trying to use that instead
